tests/streamlit_ui_tests/streamlit.spec.ts

```diff
@@ -300,6 +380,220 @@ test.describe('Home Tab Buttons', () => {
     await expect(mito.locator('p.select-text').getByText('Column2')).toBeVisible();
   })
 
+  test('Update Graph Title', async ({ page }) => {
+    const mito = await getMitoFrameWithTypeCSV(page);
+
+    await clickButtonAndAwaitResponse(page, mito, { name: 'Graph' })
+    await expect(mito.getByText('Column1 bar chart')).toBeVisible();
+
+    await mito.getByText('Column1 bar chart').dblclick();
+
+    await expect(mito.locator('.popup-input')).toBeVisible();
+    await mito.locator('.popup-input').fill('My Graph Title');
+    await mito.locator('.popup-input').press('Enter');
+
+    await awaitResponse(page);
+    await expect(mito.getByText('My Graph Title')).toBeVisible();
+  });
+
+  test('Update X axis Title on double click', async ({ page }) => {
+    const mito = await getMitoFrameWithTypeCSV(page);
+
+    await clickButtonAndAwaitResponse(page, mito, { name: 'Graph' })
+    await expect(mito.getByText('Column1 bar chart')).toBeVisible();
+
+    await mito.locator('.g-xtitle', { hasText: 'count' }).dblclick();
+
+    await expect(mito.locator('.popup-input')).toBeVisible();
+    await mito.locator('.popup-input').fill('X axis Title');
+    await mito.locator('.popup-input').press('Enter');
+
+    await awaitResponse(page);
+    await expect(mito.locator('.g-xtitle', { hasText: 'X axis Title' })).toBeVisible();
+  });
+
+  test('Update Y axis Title with double click', async ({ page }) => {
```

Comment:
Can we add a test for clicking on the legend and then updating the Y axis title by double clicking on the axis label works. Since we saw that as an issue earlier?